planner/core/fragment.go

```diff
@@ -252,17 +252,16 @@ func (e *mppTaskGenerator) generateMPPTasksForFragment(f *Fragment) (tasks []*kv
 		for _, r := range f.ExchangeReceivers {
 			childrenTasks = append(childrenTasks, r.Tasks...)
 		}
-		if f.singleton {
-			childrenTasks = childrenTasks[0:1]
+		if f.singleton && len(childrenTasks) > 0 {
+			if len(childrenTasks) > 0 {
```

Comment:
--

Reply:
the `double if` is for somehow readability... qwq

Reply:
fixed